src/v1/internal/connection-providers.js

```diff
@@ -45,38 +46,40 @@ class ConnectionProvider {
 
 export class DirectConnectionProvider extends ConnectionProvider {
 
-  constructor(hostPort, connectionPool, driverOnErrorCallback) {
+  constructor(address, connectionPool, driverOnErrorCallback) {
     super();
-    this._hostPort = hostPort;
+    this._address = address;
     this._connectionPool = connectionPool;
     this._driverOnErrorCallback = driverOnErrorCallback;
   }
 
   acquireConnection(mode) {
-    const connectionPromise = this._connectionPool.acquire(this._hostPort);
+    const connectionPromise = this._connectionPool.acquire(this._address);
     return this._withAdditionalOnErrorCallback(connectionPromise, this._driverOnErrorCallback);
   }
 }
 
 export class LoadBalancer extends ConnectionProvider {
 
-  constructor(hostPort, routingContext, connectionPool, loadBalancingStrategy, hostNameResolver, driverOnErrorCallback, log) {
+  constructor(address, routingContext, connectionPool, loadBalancingStrategy, hostNameResolver, driverOnErrorCallback, log) {
     super();
-    this._seedRouter = hostPort;
+    this._seedRouter = address;
     this._routingTable = new RoutingTable([this._seedRouter]);
     this._rediscovery = new Rediscovery(new RoutingUtil(routingContext));
     this._connectionPool = connectionPool;
     this._driverOnErrorCallback = driverOnErrorCallback;
     this._loadBalancingStrategy = loadBalancingStrategy;
     this._hostNameResolver = hostNameResolver;
+    this._dnsResolver = new HostNameResolver();
     this._log = log;
-    this._useSeedRouter = false;
+    this._useSeedRouter = true;
   }
 
   acquireConnection(accessMode) {
+    let that = this;
     const connectionPromise = this._freshRoutingTable(accessMode).then(routingTable => {
       if (accessMode === READ) {
-        const address = this._loadBalancingStrategy.selectReader(routingTable.readers);
+        const address = that._loadBalancingStrategy.selectReader(routingTable.readers);
```

Comment:
Is variable `that` needed here? It looks like a reference copy to me.

Reply:
definitely not :) thanks for the catch.